### DISCUSSION TITLE
TextFormatter / FormatLine

### DISCUSSION THREAD

**Discussion by tomhornyak** [AvaloniaUI/Avalonia#13397] [General]
I am looking for examples on how to use the TextFormatter. I have a WPF application using a RichTextBox and I would like to port that to MacOs. But I cannot find any good samples of how to format text. Thank you.

**Comment by timunie** [comment]
RichTextBox is not implemented yet but planned. We have SelectableTextBlock which Accepts formatted text

**Reply by LeoYang06** [reply]
Are there any updates on the implementation plan for FlowDocument and RichTextBox? I have a WPF editor that uses RichTextBox and I'm looking to port it to Avalonia. Thank you.

**Reply by thevortexcloud** [reply]
I can't directly answer your question but something in here might give you some hints on the best way to do it.

https://github.com/AvaloniaUI/AvaloniaEdit/

**Reply by LeoYang06** [reply]
It seems that the TextDocument in AvaloniaEditor only supports text. I need to insert arbitrary flow content elements, such as images and other controls.

**Reply by maxkatz6** [reply]
AvaloniaEdit does support inline controls, they have couple of samples with buttons embedded in the text. You can use Image control in the same way.

**Reply by LeoYang06** [reply]
Thank you. I'll look into AvaloniaEdit again.
